### PR TITLE
Taxonomystore

### PR DESCRIPTION
**Pull request by jaypal412**
## Motivation and Context

Product Search Functionlity
Product store.
## Describe your changes

Added the product Store as well as the product search.

Checkbox alignment 
- Now we have removed the space between text and  checkbox throughut  application.
## Any further comments?

------

## Checklist

- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

------
Dear Gatekeeper,

Please make sure that the commits that will be merged or rebased into the base
branch are [formatted according to our standards][commit-style].

> The only additional requirement is that the the PR number must appear in the
> commit title in brackets: `(#XXX)`

[commit-style]: https://github.com/aviabird/snitch/blob/develop/CONTRIBUTING.md#styling

### CODE REVIEW

File: app/app-routing.module.ts [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+	{ path: "search", loadChildren: "./search/search.module#SearchModule" },
+	{ path: "featured", loadChildren: "./featured/featured.module#FeaturedModule" },
+	{ path: "settings", loadChildren: "./settings/settings.module#SettingsModule" },
+	{ path: "product", loadChildren: "./product/product.module#ProductModule" }
 ];
-
 @NgModule({
-    imports: [NativeScriptRouterModule.forRoot(routes)],
-    exports: [NativeScriptRouterModule]
+	imports: [NativeScriptRouterModule.forRoot(routes)],
```
Comment (pkrawat1) [inline app/app-routing.module.ts]: tab size is 8 here, use 2 spaces

Comment (jaypal412) [reply app/app-routing.module.ts]: Done

File: app/app.module.ts [hunk 2]
```diff
@@ -1,39 +1,58 @@
...
-    declarations: [
-        AppComponent
-    ],
-    schemas: [
-        NO_ERRORS_SCHEMA
-    ],
-    providers: [
-        TaxonomyService
-    ]
+	bootstrap: [
```
Comment (ashish173) [inline app/app.module.ts]: @jaypal412 tab spaces is wrong here.

File: app/core/services/product.service.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	 * @param {string} id
+	 * @returns {Observable<Product>}
+	 *
+	 * @memberof ProductService
+	 */
+
+	getProduct(id: string): Observable<Product> {
+		return this.http
+			.get<Product>(
+				`https://app.ofypets.com/api/v1/products/${id}?data_set=large`
```
Comment (ashish173) [inline app/core/services/product.service.ts]: @jaypal412 hardcoded url's? @pkrawat1

File: app/core/services/product.service.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+				`https://app.ofypets.com/api/v1/products?q[s]=avg_rating+desc&page=${pageNumber}&per_page=20&data_set=small`
+			)
+	}
+
+
+
+	getproductsByKeyword(keyword: string): Observable<any> {
+		return this.http
+			.get<Array<Product>>(
+				`https://app.ofypets.com/api/v1/products?${keyword}&per_page=20&data_set=small`
```
Comment (ashish173) [inline app/core/services/product.service.ts]: @jaypal412 again harcoded urls.

File: app/core/services/taxonomy.service.ts [hunk 5]
```diff
@@ -4,40 +4,18 @@ import { Observable as RxObservable } from "rxjs";
 
 @Injectable()
 export class TaxonomyService {
-  private serverUrl = "https://httpbin.org/get";
+	private serverUrl = "https://httpbin.org/get";
```
Comment (ashish173) [inline app/core/services/taxonomy.service.ts]: @jaypal412 what is this used for?

File: app/featured/action/tutorial.actions.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+
+export class AddTutorial implements Action {
+	readonly type = ADD_TUTORIAL
+
+	constructor(public payload: Tutorial) { }
+}
+
+export class RemoveTutorial implements Action {
+	readonly type = REMOVE_TUTORIAL
```
Comment (ashish173) [inline app/featured/action/tutorial.actions.ts]: indentation problems.

File: app/app.module.ts (hunk 2 above)
Comment (jaypal412) [reply app/app.module.ts]: Done

File: app/core/services/product.service.ts (hunk 3 above)
Comment (pkrawat1) [reply app/core/services/product.service.ts]: @ashish173 will add them to the backlog, as will require interceptor logic. @jaypal412  Will add that in next PR

**Pull request merged**